src/sss_client/ssh/sss_ssh_knownhostsproxy.c

```diff
@@ -311,20 +311,27 @@ int main(int argc, const char **argv)
         ret = connect_proxy_command(discard_const(pc_args));
     } else if (ai) {
         /* Try all IP addresses before giving up */
+        int socket_descriptor = -1;
         for (struct addrinfo *ti = ai; ti != NULL; ti = ti->ai_next) {
-            int socket_descriptor = -1;
             ret = connect_socket(ti->ai_family, ti->ai_addr, ti->ai_addrlen,
                                  &socket_descriptor);
-            if (ret == 0) {
-                ret = proxy_data(socket_descriptor);
+            if (ret == EOK) {
                 break;
             }
         }
+
+        if (ret == EOK) {
+            ret = proxy_data(socket_descriptor);
```

Comment:
Wouldn't it make sense to print error message here as well (to have all failure cases covered)?

Reply:
Added.